### ISSUE TITLE
ZigZag UnhandledException with small data sets

### ISSUE THREAD

**Issue by Fredrik-C** [DaveSkender/Stock.Indicators#632]
[zigzag_issue.zip](https://github.com/DaveSkender/Stock.Indicators/files/7573199/zigzag_issue.zip)
When dataset is small and does not have enough volatility to generate enough zigzag data then exception is thrown.

Error message(s):

```bash
System.ArgumentOutOfRangeException
Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at Skender.Stock.Indicators.Indicator.DrawRetraceLine(List`1 results, String lastDirection, ZigZagPoint lastLowPoint, ZigZagPoint lastHighPoint, ZigZagPoint nextPoint)
   at Skender.Stock.Indicators.Indicator.GetZigZag[TQuote](IEnumerable`1 quotes, EndType endType, Decimal percentChange)
```

## to reproduce
Use attached sample file.
With default zigzag arguments exception is thrown.
However, using "GetZigZag(EndType.Close, 0.1m)" is fine.

## expected behavior

A validation error or an return dataset without zigzag data populated.

**Comment by DaveSkender** [comment]
Thanks for reporting.  Will fix quickly.

**Comment by DaveSkender** [comment]
Looks like this data does present a confirmed low point with a +5% reversal; however, the last code update (#618) failed to accommodate drawing of the retrace lines properly when only one line segment exists.

![image](https://user-images.githubusercontent.com/8432125/142710478-c5863684-b722-41b5-a103-cc85fc2f44d3.png)

**Comment by DaveSkender** [comment]
This be was introduced in version 1.19.1 and resolved in version 1.19.2
